sympy/functions/special/tests/test_error_functions.py

```diff
@@ -437,6 +437,11 @@ def test_ei():
     assert Ei(x).series(x, 1, 3) == Ei(1) + E*(x - 1) + O((x - 1)**3, (x, 1))
     assert Ei(x).series(x, oo) == \
         (120/x**5 + 24/x**4 + 6/x**3 + 2/x**2 + 1/x + 1 + O(x**(-6), (x, oo)))*exp(x)/x
+    assert Ei(x).series(x, -oo) == \
+        (120/x**5 + 24/x**4 + 6/x**3 + 2/x**2 + 1/x + 1 + O(x**(-6), (x, -oo)))*exp(x)/x
+    assert Ei(-x).series(x, oo).expand() == \
+        120*exp(-x)/x**6 - 24*exp(-x)/x**5 + 6*exp(-x)/x**4 - 2*exp(-x)/x**3 + \
+        exp(-x)/x**2 - exp(-x)/x + O(exp(-x)/x**7, (x, oo))
```

Comment:
Why is `expand` being called in one case but not the other?
```python
In [1]: Ei(-x).series(x, oo)
Out[1]: 
 ⎛  120   24   6    2    1        ⎛1        ⎞⎞  -x 
-⎜- ─── + ── - ── + ── - ─ + 1 + O⎜──; x → ∞⎟⎟⋅ℯ   
 ⎜   5     4    3    2   x        ⎜ 6       ⎟⎟     
 ⎝  x     x    x    x             ⎝x        ⎠⎠     
───────────────────────────────────────────────────
                         x                         

In [2]: Ei(-x).series(x, -oo)
Out[2]: 
 ⎛  120   24   6    2    1        ⎛1         ⎞⎞  -x 
-⎜- ─── + ── - ── + ── - ─ + 1 + O⎜──; x → -∞⎟⎟⋅ℯ   
 ⎜   5     4    3    2   x        ⎜ 6        ⎟⎟     
 ⎝  x     x    x    x             ⎝x         ⎠⎠     
────────────────────────────────────────────────────
                         x                          

In [3]: Ei(-x).series(x, -oo).expand()
Out[3]: 
     -x       -x      -x      -x    -x    -x    ⎛  -x         ⎞
120⋅ℯ     24⋅ℯ     6⋅ℯ     2⋅ℯ     ℯ     ℯ      ⎜-ℯ           ⎟
─────── - ────── + ───── - ───── + ─── - ─── + O⎜─────; x → -∞⎟
   6         5       4       3      2     x     ⎜  7          ⎟
  x         x       x       x      x            ⎝ x           ⎠
```

Reply:
Well cause I wasn't able to figure out why something like fails
```
>>> Ei(-x).series(x, oo)
-(-120/x**5 + 24/x**4 - 6/x**3 + 2/x**2 - 1/x + 1 + O(x**(-6), (x, oo)))*exp(-x)/x
>>> -(-120/x**5 + 24/x**4 - 6/x**3 + 2/x**2 - 1/x + 1 + O(x**(-6), (x, oo)))*exp(-x)/x
(120/x**5 - 24/x**4 + 6/x**3 - 2/x**2 + 1/x - 1 + O(x**(-6), (x, oo)))*exp(-x)/x
>>> assert Ei(-x).series(x, oo) == (120/x**5 - 24/x**4 + 6/x**3 - 2/x**2 + 1/x - 1 + O(x**(-6), (x, oo)))*exp(-x)/x
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
```

And hence wasn't sure how could I add a test for this and hence I ended up using expand !

Reply:
Brackets are needed:
```python
In [3]: -(-120/x**5 + 24/x**4 - 6/x**3 + 2/x**2 - 1/x + 1 + O(x**(-6), (x, oo)))*exp(-x)/x
Out[3]: 
⎛120   24   6    2    1        ⎛1        ⎞⎞  -x
⎜─── - ── + ── - ── + ─ - 1 + O⎜──; x → ∞⎟⎟⋅ℯ  
⎜ 5     4    3    2   x        ⎜ 6       ⎟⎟    
⎝x     x    x    x             ⎝x        ⎠⎠    
───────────────────────────────────────────────
                       x                       

In [4]: -((-120/x**5 + 24/x**4 - 6/x**3 + 2/x**2 - 1/x + 1 + O(x**(-6), (x, oo)))*exp(-x)/x)
Out[4]: 
 ⎛  120   24   6    2    1        ⎛1        ⎞⎞  -x 
-⎜- ─── + ── - ── + ── - ─ + 1 + O⎜──; x → ∞⎟⎟⋅ℯ   
 ⎜   5     4    3    2   x        ⎜ 6       ⎟⎟     
 ⎝  x     x    x    x             ⎝x        ⎠⎠     
───────────────────────────────────────────────────
                         x 
```
This is because of automatic distribution in a 2-arg Mul (a longstanding issue):
```python
In [5]: 2*(x + y)*z
Out[5]: z⋅(2⋅x + 2⋅y)

In [6]: 2*((x + y)*z)
Out[6]: 2⋅z⋅(x + y)
```

Reply:
Thanks for the help !

Reply:
Probably in this case distribution is the right thing though and the series code should be changed to return the distributed result.

Reply:
Arghh I just committed the change with the brackets :|

Reply:
That's fine. This PR is good. Just a future PR could maybe handle this. Ideally we want the series to be in a well-defined canonical form so that tests like this are not so fragile.